### DISCUSSION TITLE
How can I control the activity name for the root activity?

### DISCUSSION THREAD

**Discussion by ithielnor** [open-telemetry/opentelemetry-dotnet#3282] [Q&A] [answered]
It looks like `AddAspNetInstrumentation` `Enrich` option supports changing the activity DisplayName, but not the actual name. It appears that DisplayName is set to the same value as the http.target tag, while the activity Name is set to the same value as http.route. Is there some means I can use to control how the actual name of the activity is generated? 

I'm using the ASP.NET http module in .Net Framework 4.7.2.

**Comment by cijothomas** [comment, marked as answer]
> changing the activity DisplayName, but not the actual name

Can you clarify what is meant by "ActualName"?
Activity.DisplayName is the SpanName from OTel specification.

**Reply by ithielnor** [reply]
My mistake. In testing I had tried setting the DisplayName in the OnStartActivity event, but I think it get's overwritten later. Setting it in the OnStopActivity event seems to work.
